tests/dialects/test_dlti.py

```diff
@@ -0,0 +1,264 @@
+from collections.abc import Mapping
+from typing import TypeAlias
+
+import pytest
+
+from xdsl.dialects.builtin import (
+    ArrayAttr,
+    Float32Type,
+    FloatAttr,
+    IntAttr,
+    StringAttr,
+    i32,
+)
+from xdsl.dialects.dlti import (
+    DataLayoutEntryAttr,
+    DataLayoutSpecAttr,
+    DLTIEntryMap,
+    MapAttr,
+    TargetDeviceSpecAttr,
+    TargetSystemSpecAttr,
+)
+from xdsl.ir import Attribute, TypeAttribute
+from xdsl.utils.exceptions import VerifyException
+from xdsl.utils.hints import isa
+
+DictValueType: TypeAlias = Mapping[
+    StringAttr | TypeAttribute | str, "Attribute | str | int | DictValueType"
+]
+
+
+def test_data_layout_entry():
+    # Test passing strings for key and value
+    entry = DataLayoutEntryAttr("test", "V")
+    assert isinstance(entry.key, StringAttr)
+    assert isinstance(entry.value, StringAttr)
+    assert entry.key.data == "test"
+    assert entry.value.data == "V"
```

Comment:
Could you please condense the tests a little bit now that we've iterated on implementation? Could you please test directly for key/value below instead of unwrapping data?
```suggestion
    # String key and value
    entry = DataLayoutEntryAttr("test", "V")
    assert entry.key == StringAttr("test")
    assert entry.value == StringAttr("V")
```

Reply:
Ah, nice - I have done this, make it much shorter :)